README.md

```diff
@@ -79,6 +79,65 @@ cd corral
 
 You can make a debug build with `.\make.ps1 build -Config Debug`
 
+## Getting started using Corral
+After instllation, make sure Corral's current path is also set in your PATH environment variable. This will make it easier by not entering the full path each time when you use one of Corral commands in your projects.
+
+Follow these steps to create your first project using Corral.
+
+1. Create an empty folder and switch to this directory. This will be our example project to use Corral
+    ```bash
+    mkdir myproject
+    cd myproject
+    ```
+
+2. Initialize Corral. It will create `corral.json` and `lock.json` files. At this moment they won't have much information since we haven't added any dependency yet.
+    ```bash
+    corral init
+    ```
+
+3. Add a dependency. This is the way to tell Corral that your project depends on this and you want to include it when building your project.
+    ```bash
+    corral add github.com/jemc/pony-inspect.git
+    ```
+
+4. Use that dependency in your code. Create a file `main.pony` with following code.
+    ```pony
+    use "inspect"
+      actor Main
+        new create(env: Env) =>
+          env.out.print(Inspect("Hello, World!"))
+    ```
+
+5. Fetch dependencies. As you can see in previous step that we are using `Inspect` type from the added depedency, we need to make sure that the code for this should be locally available on your machine so that we can compile without any errors.
+    ```bash
+    corral fetch
+    ```
+
+6. Build the project. Corral will now use all of the information which we have provide and try to build the project.
+    ```bash
+    corral run ponyc
+    ```
+
+If there are no errors generated then an executable `myproject` will be created in the same folder.
+
+If you prefer you can just copy the following code and run it in your favourite shell which will basically do the same which we have explained in previous steps.
+
+```bash
+mkdir myproject && cd myproject
+corral init
+corral add github.com/jemc/pony-inspect.git
+
+echo '
+use "inspect"
+actor Main
+  new create(env: Env) =>
+    env.out.print(Inspect("Hello, World!"))
+' > main.pony
+
+corral fetch
+corral run ponyc
+```
+
 ## Create a project with dependencies
```

Comment:
This header now feels very odd.

It made sense before the new additions.

Really what we have here is we are starting more detailed instructions for using corral. This is the beginning of more in-depth documentation, but that isn't indicated. I think it makes to move this to a new file DOCS.md that has a top-level.

# Documentation 

and then the rest of the content from here to the end.

Then, the getting started section would end with a "to learn more about Corral and it's features, see the documentation" with a link.

I think that also having a section here

```
# Documentation

see [Docs.md](DOCS.md).
```

would be good.

Reply:
I second the section header change. Moving this to a separate file would also allow more reorganization into different concerns such as taking the sentence on versioning and making a distinct dependency versioning section that can further explain how corral resolves these -- see the classic diamond problem which should be in the documentation somewhere if not close to versioning of dependencies as I suggest here.